extension/js/common/core/msg-block-parser.ts

```diff
@@ -90,6 +90,9 @@ export class MsgBlockParser {
     // thus we use RegEx so that it works on both browser and node
     if (decryptedContent.includes('class="cryptup_file"')) {
       decryptedContent = decryptedContent.replace(/<a\s+href="([^"]+)"\s+class="cryptup_file"\s+cryptup-data="([^"]+)"\s*>[^<]+<\/a>\n?/gm, (_, url, fcData) => {
+        if (!decryptedContent.match(/<a\s+href="(https?:\/\/flowcrypt\.s3\.amazonaws\.com\/[^"]+)"\s+class="cryptup_file"\s+cryptup-data="([^"]+)"\s*>[^<]+<\/a>\n?/gm)) {
```

Comment:
Don't need to call the regex.match() twice.
Please check `url` parameter here. It has the value of what is captured inside of `href="..."`
It's ok to extract the host from there like you previously did.